src/file.cpp

```diff
@@ -309,4 +309,35 @@ namespace zim
 
     return ret;
   }
+
+  bool validate(const std::string& zimPath)
+  {
+      try {
+          std::vector<std::string> urlList;
+          std::vector<std::string> titleList;
+
+          zim::File f(zimPath);
+
+          auto nrArticles = f.getCountArticles();
+
+          urlList.reserve(nrArticles);
+          titleList.reserve(nrArticles);
+
+          for (zim::article_index_type i = 0; i < nrArticles; i++)
+          {
+              auto article = f.getArticle(i);
```

Comment:
Getting a article using its index will use the `UrlPtrList` to get the article. So the article should indeed be url sorted (or url increasing as index is increasing).

But it say nothing about the title. Even on a correct zim file, the title order may differ from the url order.
This is way we have the TitlePtrList.

- `Dirent.atOffset(UrlPtrList[idx]).fullurl` should be increasing when idx is increasing
- `Dirent.atOffset(UrlPtrList[TitlePtrList[idx]]).fulltitle` should be increasing when idx is increasing.

(`atOffset` and `fulltitle` don't exist but you got the idea :) )